### ISSUE TITLE
Add Typescript types for SMS Status Callbacks

### ISSUE THREAD

**Issue by emagaram** [twilio/twilio-node#802]
I'm pretty sure there are no typescript types for SMS Status Callback Webhooks. Could you add this in the future? Thanks!

**Comment by beebzz** [comment]
Hi @emagaram, thanks for posting an issue! Could you specify where this typing is missing? I see that status callbacks for SMS messages are typed as strings [here](https://github.com/twilio/twilio-node/blob/main/lib/rest/api/v2010/account/message.d.ts#L189).

**Comment by emagaram** [comment]
I was referring to the HTTP body that the webhook sends. Something like this:

```
type SMSStatusWebhook = {
  SmsSid: string;
  SmsStatus: "accepted" | "scheduled" | "queued" | "sending" | "sent";
  MessageStatus: string;
  To: string;
  MessagingServiceSid: string;
  MessageSid: string;
  AccountSid: string;
  From: string;
  ApiVersion: string;
};
```

**Comment by PCOffline** [comment]
Bumping this, also I defined some types myself when using Status Callbacks:
```ts
export enum SmsStatuses {
    sent = 'SENT',
    delivered = 'DELIVERED',
    undelivered = 'UNDELIVERED',
    failed = 'FAILED',
    queued = 'QUEUED',
}

export interface StatusCallback {
    AccountSid: `AC${string}`;
    MessageSid: `SM${string}`;
    SmsSid: `SM${string}`;
    originalSms: `SM${string}`;
    MessageStatus: SmsStatuses;
    SmsStatus: SmsStatuses;
    From: `+${number}`;
    To: `+${number}`;
    ApiVersion: `${number}-${number}-${number}`;
}
```

**Comment by PCOffline** [comment]
There are a few more statuses like 'accepted' and 'scheduled' that I am not sure where to place but this is the gist of it

**Comment by PCOffline** [comment]
I am not sure if there are differences in fields' values like SmsSid vs MessageSid, MessageStatus and SmsStatus etc. but that's up to the reviewers

**Comment by vinicius73** [comment]
Previously, in v3 `MessageStatus` was exported.
Now in v4, is not exported anymore.

There is a reason for that?

https://github.com/twilio/twilio-node/commit/b471067195b2e67d8073d70db6432cb14609fedb#diff-2adf184f135ba5d85051c5b5f187adf3d9eb878362c5a04be926ef951b1ff5afL477

**Comment by Hunga1** [comment]
Hi @vinicius73. Thanks for pointing this out. We'll add this issue to our backlog to be prioritized (internal reference: DI-2576). +1s and comments on the issue summary will help it move up the backlog.

In the future, please open a new issue thread so that we can better triage the issue, help to avoid potentially missing your comment, and help track the progress individually from the OP.

**Comment by PCOffline** [comment]
And any news about adding Status Callback type, as per the suggestion above or in general?

**Comment by childish-sambino** [comment]
@PCOffline We have this on our internal backlog (DI-2383).

**Comment by vinicius73** [comment]
Thanks @Hunga1

**Comment by claudiachua** [comment]
This issue has been added to our internal backlog to be prioritized as it changes generated configuration. +1s on the issue summary will help it move up the backlog (DI-2383).

**Comment by castdin** [comment]
It would of course be quite nice if Twilio provided types for *all* the different webhook request bodies, including SMS and Voice. This issue has a common thread with #588 which I see has also been marked as `DI-2383` on Twilio's internal backlog.

Can anyone (perhaps @claudiachua) speak to the current status of DI-2383? How far down the backlog is it? I would hope Twilio considers DX a priority. 🙂

**Comment by PCOffline** [comment]
Yeah this issue has been stale for over a year, when the change is small enough to take a day of work for me, which would include both research about every field and actual code, so I'd assume it'd take even less time for a Twilio employee familiar with the codebase and API.

**Comment by PCOffline** [comment]
Hey, @Hunga1, @claudiachua, any updates on this? I don't see a reason why such a small change would take over a year.

**Comment by otowncaleb** [comment]
> Yeah this issue has been stale for over a year. The change is small enough to take a day of work for me, which would include both research about every field and actual code, so I'd assume it'd take even less time for a Twilio employee familiar with the codebase and API.

@PCOffline Would you perhaps have the bandwidth to submit a PR? I wonder if Twilio would take it given the right conditions.

**Comment by PCOffline** [comment]
@otowncaleb ill try :)

**Comment by satishcoder20** [comment]
Are you doing international calls using Twilio?
Here’s a free Chrome extension where you can connect your Twilio account and start making calls and sending SMS instantly.
It’s completely free and secure — no data is stored on any server.
Everything runs locally on your chrome browser.

Extension link: https://chromewebstore.google.com/detail/fpcemcdkoaflldcailmbbnnahnnckhjc
